playbooks/azure/openshift-cluster/create_offer.yml

```diff
@@ -0,0 +1,83 @@
+---
+- hosts: localhost
+  gather_facts: no
+  tasks:
+  - name: ensure libraries are available
+    import_role:
+      name: lib_utils
+  
+  - name: set expire date
+    set_fact:
+      sas_expire: "{{ lookup('pipe', 'date -d \"14 days\" +\"%Y-%m-%dT%H:%M:00Z\"') }}"
+      disk: "{{ image_name }}.vhd"
+  
+  - name: fetch the image for the tags
+    command: |
+      az image list --resource-group images  -o json --query "[?name=='{{ image_name }}'].tags"
+    register: imagetags
+  
+  - name: generate storage blob url
+    command: |
+      az storage blob url -c {{ openshift_azure_container }} --account-name {{ openshift_azure_storage_account }} -n {{ disk }} --output=json
+    register: bloburl
+
+  - name: generate sas url for the container
+    command: |
+      az storage container generate-sas --name {{ openshift_azure_container }} --account-name {{ openshift_azure_storage_account }} --permissions rl --expiry {{ sas_expire }} --output=json
+    register: sasurl
+  
+  - name: set the sas URLS
+    set_fact:
+      openshift_azure_sas_url: "{{ bloburl.stdout|from_json + '?' + sasurl.stdout|from_json }}"
+  
+  - name: set the image tags
+    set_fact:
+      image_tags: "{{ (imagetags.stdout | from_json)[0] }}"
```

Comment:
@maxamillion this is an example of the `{{ (foo.stdout | from_json)... }}` syntax that is bugging me ;)